base/ca/src/org/dogtagpki/server/ca/rest/CertService.java

```diff
@@ -190,8 +193,13 @@ public Response revokeCert(CertId id, CertRevokeRequest request, boolean caCert)
 
             processor.setAuthority(authority);
 
+            caIssuerDN = authority.getCACert().getIssuerDN().toString();
+            caX500DN = new X500Name(caIssuerDN);
```

Comment:
getCACert() returns X509CertImpl in JSS, where getIssuerDN() is defined as the following:
public Principal getIssuerDN()

Reply:
Yes, `X500Name` implements `Principal`. If we want to know the actual type we could either analyze the source to see how the issuer DN was set, or run some tests and check the actual class of the return value. But if the encoding is not a concern we can leave the code as is, so it will convert the issuer DN (whatever that is) into `String` then to `X500Name`, or we can just leave it as `String`.

Reply:
yes it's best to preserve the encoding.  Thanks.